src/extension.ts

```diff
@@ -3,139 +3,120 @@
-              options
-            );
-
-            childProcess.stdout.on("data", (data) => {
-              console.log(`stdout: ${data}`);
-            });
-
-            childProcess.stderr.on("data", (data) => {
-              console.error(`stderr: ${data}`);
-            });
+        // grab a random port.
+        server.listen(() => {
+          // Start the child java process
+          let options = { cwd: context.extensionPath };
+
+          let port = (server.address() as net.AddressInfo).port;
+
+          let version = vscode.workspace
+            .getConfiguration("smithyLsp")
+            .get("version", "`");
+
+          // Downloading latest poms
+          let resolveArgs = [
+            "resolve",
+            "--mode",
+            "force",
+            "com.disneystreaming.smithy:smithy-language-server:" + version,
+            "-r",
+            "m2local",
+          ];
+          let resolveProcess = child_process.spawn(
+            executable,
+            resolveArgs,
+            options
+          );
+          resolveProcess.on("exit", (exitCode) => {
+            console.log("Exit code : " + exitCode);
+            if (exitCode == 0) {
+              console.log(
+                "Launching smithy-language-server version:" + version
+              );
+
+              let launchargs = [
+                "launch",
+                "com.disneystreaming.smithy:smithy-language-server:" + version,
+                "-r",
+                "m2local",
+                "--",
+                port.toString(),
+              ];
+
+              let childProcess = child_process.spawn(
+                executable,
+                launchargs,
+                options
+              );
+
+              childProcess.stdout.on("data", (data) => {
+                console.log(`stdout: ${data}`);
+              });
+
+              childProcess.stderr.on("data", (data) => {
+                console.error(`stderr: ${data}`);
+              });
+
+              childProcess.on("close", (code) => {
+                console.log(`LSP exited with code ${code}`);
+              });
+            } else {
+              console.log(
+                `Could not resolve smithy-language-server implementation`
+              );
+            }
+          });
 
-            childProcess.on("close", (code) => {
-              console.log(`LSP exited with code ${code}`);
-            });
-          } else {
-            console.log(
-              `Could not resolve smithy-language-server implementation`
-            );
-          }
+          // Send raw output to a file
+          if (!fs.existsSync(context.storagePath))
+            fs.mkdirSync(context.storagePath);
         });
-
-        // Send raw output to a file
-        if (!fs.existsSync(context.storagePath))
-          fs.mkdirSync(context.storagePath);
       });
-    });
+    }
+
+    return getCoursierExecutable(context.extensionPath).then((binaryPath) =>
```

Comment:
use `context.globalStorageUri ` here instead. 

See https://code.visualstudio.com/api/references/vscode-api#ExtensionContext.globalStorageUri

Reply:
done in [27ae236](https://github.com/disneystreaming/vscode-smithy/pull/5/commits/27ae236dbfe6049a81852cc8ccf154f8d40ee5bb)